src/sampleutil.h

```diff
@@ -43,6 +43,13 @@
 // A group of utilities for working with samples.
 class SampleUtil {
   public:
+    enum CLIPPING {
+        NO_CLIPPING = 0,
+        CLIPPING_LEFT = 1,
+        CLIPPING_RIGHT = 2,
+        CLIPPING_BOTH = 3,
```

Comment:
explicit bitmask sounds good:

CLIPPING_CH1 = 1
CLIPPING_CH2 = 2
seems fine, and then future:
CLIPPING_CH3 = 4
CLIPPING_CH4 = 8